clients/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/NessieUtils.scala

```diff
@@ -164,22 +164,30 @@ object NessieUtils {
       catalog: Option[String]
   ): NessieApiV1 = {
     val catalogName = catalog.getOrElse(currentCatalog.name)
-    val catalogConf = SparkSession.active.sparkContext.conf
+    val sparkConf = SparkSession.active.sparkContext.conf
+    val catalogConf = sparkConf
       .getAllWithPrefix(s"spark.sql.catalog.$catalogName.")
       .toMap
-    // Referring to https://github.com/apache/iceberg/blob/master/nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java
-    // Not using fully-qualified class name to provide protection from shading activities (if any)
-    val catalogImpl = catalogConf.get("catalog-impl")
-    val catalogInfo = catalogImpl match {
-      case Some(clazz) => s"but $catalogName is a $clazz"
-      case None =>
-        s"but spark.sql.catalog.$catalogName.catalog-impl is not set"
+
+    val catalogClass = sparkConf.getOption(s"spark.sql.catalog.$catalogName")
+    val needsImplCheck =
+      catalogClass.map(!_.endsWith(".DeltaCatalog")).getOrElse(true)
+    if (needsImplCheck) {
+      val catalogImpl = catalogConf.get("catalog-impl")
+      val catalogErrorDetail = catalogImpl match {
+        case Some(clazz) => s"but $catalogName is a $clazz"
+        case None =>
+          s"but spark.sql.catalog.$catalogName.catalog-impl is not set"
+      }
+      // Referring to https://github.com/apache/iceberg/blob/master/nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java
+      // Not using fully-qualified class name to provide protection from shading activities (if any)
+      require(
+        catalogImpl
+          .exists(impl => impl.endsWith(".NessieCatalog")),
+        s"The command works only when the catalog is a NessieCatalog ($catalogErrorDetail). Either set the catalog via USE <catalog_name> or provide the catalog during execution: <command> IN <catalog_name>."
```

Comment:
Note that we have tests to make sure this still works in the iceberg case:
https://github.com/projectnessie/nessie/blob/8ba7079c54325c90e39ebf4e234c014711ca0e8e/clients/spark-extensions-base/src/test/java/org/projectnessie/spark/extensions/AbstractSparkSqlTest.java#L703-L721

I could not use the same style of testing for delta because i didnt know which non-delta catalog to use (no iceberg jar on classpath).